### ISSUE TITLE
[calendar] disabledDates on month mode/type

### ISSUE THREAD

**Issue by unbreak** [fomantic/Fomantic-UI#1086]
# Help Wanted
Is any way set disabledDates on calendar in month mode?

## Problem
I've tried set:
```
type: "month",
disabledDates: [{date: new Date("2019-04"), message: "Some message"},{date: new Date("2019-05-01"), message: "Some other message"}]
```

But it didn't work, when i switch type to `day` it woks fine :)

**Comment by hammy2899** [comment]
Currently you can't simply disable a month or year however I think we could easily do this by adding `month` and `year` options to the `disabledDates` object.

Example:
```javascript
[
  {
    date: new Date('2019-10-11'),
    message: 'Message'
  },
  {
    month: 11, // November
               // could also accept Date and get month from it `date.getMonth() + 1`
    message: 'Message'
  },
  {
    year: 2020,
    message: 'Message'
  }
]
```

This logic would need to be added to the `isDisabled` helper function https://github.com/fomantic/Fomantic-UI/blob/master/src/definitions/modules/calendar.js#L836

**Comment by szalonna** [comment]
I would like to take that!

**Comment by szalonna** [comment]
I started to implement this but I have a question in related to the months. In opposite of years and dates, months are repeating. In dates there is the `disabledDaysOfWeek` for disabling repeated days on weekly basis, but what happens if the user set e.g. the month to a specific date or a number?

Possible implementations:
1. Mix repetitive and non-repetitive disable as:
  - if the `month` is a date, disable only that specific month
  - if the `month` is a number, disable that month in every year
2. Split them in two different properties (like `disabledDaysOfWeek`)
  - in `disabledDates` if the `month` is a date, disable that specific month and only accept dates and date strings
  - introduce a new option as `disabledMonthsOfYear`, with which you can define the disabled months in a yearly basis

**Comment by szalonna** [comment]
And another question: do you want to introduce the same behavior with the `enabledDates` rules as well?

**Comment by hammy2899** [comment]
@szalonna Sorry for late reply.

I agree with @unbreak and think the first option better suits the current implementation.

**Comment by prudho** [comment]
Implemented by #1330.
